src/time/time_format/formatter.cr

```diff
@@ -176,6 +176,101 @@ struct TimeFormat
       {negative, hours, minutes}
     end
 
+    # Compute week of the year, monday is the first day of the week
+    def week_of_year_monday_1_7
+      io << Formatter.iso8601_week_of_year(time, 1)
+    end
+
+    # Compute week of the year, sunday is the first day of the week
+    def week_of_year_sunday_0_6
+      io << Formatter.iso8601_week_of_year(time, 0)
+    end
+
+    # Internal helper wethod: Figure how many weeks into the year
+    def self.week_of_year(time, firstweekday)
+      day_of_week = TimeFormat.new("%w").format(time).to_i
+      week_num = 0
+
+      if firstweekday == 1
+        if day_of_week == 0 # sunday
+          day_of_week = 6
+        else
+          day_of_week -= 1
+        end
+      end
+
+      day_of_year = TimeFormat.new("%j").format(time).to_i
+      week_num = ((day_of_year + 7 - day_of_week) / 7)
+      week_num = 0 if week_num < 0
+
+      week_num
+    end
+
+    # Compute week number according to ISO 8601
+    def self.iso8601_week_of_year(time, firstweekday)
```

Comment:
Note that if you want to make this one private you'll need to move it to the instance level since you can't call private class methods from instance methods.